### DISCUSSION TITLE
Support basic authentication when sending alert to alertmanager

### DISCUSSION THREAD

**Discussion by alexku7** [jertel/elastalert2#572] [Ideas]
In many cases the alertmanager requires authentication. Usually the alertmanager is behind some proxy with basic_auth or it can support it by itself.

Could this feature be added  ?

**Comment by nsano-rururu** [comment]
If you want that feature, please submit a pull request according to the [Contribution Guidelines](https://github.com/jertel/elastalert2/blob/master/CONTRIBUTING.md).

**Reply by nsano-rururu** [reply]
By the way, I don't understand which of HTTPBasicAuth, HTTPDigestAuth, and HTTPProxyAuth is required, so I can't.

**Reply by alexku7** [reply]
I think alertmanager supports HTTPBasicAuth only.

It should be something like this :
from requests.auth import HTTPBasicAuth

requests.get('https://api.github.com/user', auth=HTTPBasicAuth('user', 'pass'))

**Reply by nsano-rururu** [reply]
Please wait a moment as we will add `alertmanager_basic_auth_login` and `alertmanager_basic_auth_password`.

**Reply by nsano-rururu** [reply]
I submitted a pull request
https://github.com/jertel/elastalert2/pull/575
